src/case.rs

```diff
@@ -58,11 +67,11 @@ impl TestCase {
         })
     }
 
-    pub(crate) async fn execute<W>(&self, db: &dyn Database, writer: &mut W) -> Result<()>
+    pub(crate) async fn execute<W>(&mut self, db: &dyn Database, writer: &mut W) -> Result<()>
```

Comment:
Why this require mut reference?

Reply:
My thought is a pre-query interceptor may also change the input query, which requires `&mut self` because the input string is stored in `Query`.

E.g.: insert 10000 lines can be written to
```sql
-- SQLNESS BLABLA number={1..10000}
INSERT INTO TABLE t VALUES {(number)};
```

I'm not very clear with this use case, but changing this function's signature is a breaking change, so I preserve the `&mut` in advance. Do you think it's necessary?

Reply:
That sounds a good use case.